### ISSUE TITLE
IMap: getAll operation increases element TTL

### ISSUE THREAD

**Issue by vladlapayev** [hazelcast/hazelcast#5333]
Hi there,

I found a strange behavior of the IMap getAll method. I put a single value to the IMap with some ttl. Then I periodically invoke a getAll method to get access to this value while ttl is not reached. But ttl will be never reached during execution. 
Access period is less that ttl. 

Example:
```
final String key =  "key-1";
IMap<String, String> map = client.getMap("test");
Set<String> keys = new HashSet<String>() {{ add(key); add("other_key");}};

for (int i = 0; i < 20; i++) {
    Map<String, String> result = map.getAll(keys);
    if (result.isEmpty()) {
        map.put(key, "value", 500, TimeUnit.MILLISECONDS);
         System.out.println("put element");
     } else {
          System.out.println("element exists, waiting...");
     }
     Thread.sleep(100); // after 5 sleeps element must be expired
}
```

In this case element with key=key-1 will expire only when execution of this code is finished.
Code works as expected when I replace getAll by get methodl.

Hazelcast 3.4.2

**Comment by yarki** [comment]
+1

**Comment by yarki** [comment]
+1

**Comment by ahmetmircik** [comment]
Thanks for reporting this, will be fixed in 3.4.3.

**Comment by ahmetmircik** [comment]
closed via https://github.com/hazelcast/hazelcast/pull/5343
